realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java

```diff
@@ -996,4 +999,187 @@ public void realmListRemoveAllFromRealm() {
         assertEquals(0, list.size());
         assertEquals(0, realm.where(Dog.CLASS_NAME).count());
     }
+
+    // Test if waitForChange gets waked up by 1) empty transaction 2) transaction with some data changes.
```

Comment:
Split this unit test in two, one for each scenario. Right now the unit test is a bit complicated because it tries to cover both cases.